### PR TITLE
[Enhancement] Add RHEL-7 remediation for 'audit_rules_privileged_commands' rule

### PR DESCRIPTION
**Pull request by iankko**
This changeset performs the following:
* patch https://github.com/OpenSCAP/scap-security-guide/commit/478033e63056bbd08e3d2a3e0aba6fcbb21cd964 splits remediations functions library ```scap-security-guide/shared/input/fixes/bash/templates/remediation_functions``` content into to two sections (universal remediation functions and remediation functions per product),

* patch https://github.com/OpenSCAP/scap-security-guide/commit/dd3b9ad594938bbbed51e264fe53b6f65ad82ca2 introduces a new remediation function ```perform_audit_rules_privileged_commands_remediation```, which is intended to perform the remediation for ```audit_rules_privileged_commands``` rule on various products (RHEL-6, RHEL-7 etc.), Also applies this function to implement remediation for RHEL-7 system,

* patch https://github.com/OpenSCAP/scap-security-guide/commit/948cad57323a328b5206455a580ce89d5d36729c  is fixing the behaviour of the 'perform_audit_rules_privileged_commands_remediation' remediation function when run in "augenrules" mode not to produce duplicate audit rule entries for privileged commands (SUID / SGID) binaries,

and finally patch

* https://github.com/OpenSCAP/scap-security-guide/commit/96ef92e94308d48b8aa8957fe59d287f94c70a16 is rewriting the currently existing RHEL-6 remediation for the ```audit_rules_privileged_commands``` rule in order to use this new remediation function for RHEL-6 environment (min_auid = "500").

<b>Note:</b> It is a known issue (to me) that running ```shellcheck``` on ```shared/fixes/bash/templates/remediation_functions``` currently produces some warnings. AFAICT not all of them (especially those related with pkgs install) can't be blindly fixed with simple escaping the arguments with double quotes -- as far as I tested, some of the pkgs manipulation related ones stopped working when escaped. Therefore I have resigned to try to fix these warnings quickly / without further testing && will fix them somewhere in the future (once I have got more time for proper testing of each of them).

Testing report:
-------------------
AFAICT from testing the proposed rewritten remediation is working properly on both of RHEL-6 && RHEL-7 systems.

Please review.

Thank you, Jan.

### CODE REVIEW

File: shared/fixes/bash/templates/remediation_functions [hunk 1]
```diff
@@ -491,3 +460,212 @@ if ! [ "x$xinetd" = x ] ; then
...
+
+		# Search current audit rules file's content for match. Match criteria:
+		# * existing rule is for the same SUID/SGID binary we are currently processing (but
+		#   can contain multiple -F path= elements covering multiple SUID/SGID binaries)
+		# * existing rule contains all arguments from expected rule form (though can contain
+		#   them in arbitrary order)
+	
+		base_search=$(sed -e "/-a always,exit/!d" -e "/-F path=${sbinary_esc}/!d"   \
+		    		  -e "/-F path=[^[:space:]]\+/!d" -e "/-F perm=.*/!d"       \
+				  -e "/-F auid>=${min_auid}/!d" -e "/-F auid!=4294967295/!d"  \
```
Comment (jan-cerny) [inline shared/fixes/bash/templates/remediation_functions]: Why auid != 4294967295 ?

Comment (iankko) [inline shared/fixes/bash/templates/remediation_functions]: This is commonly known "constant" for audit rules syntax. From audit.rules(7) manual page:
```
The  audit  system  considers uids to be unsigned numbers. The audit system uses the number -1 to indicate that a loginuid is not set. This means that when its printed out, it looks like 4294967295. If you write a rule that you  wanted  try  to  get  the  valid  users  of  the  system,  you  need  to  look  in  /etc/login.defs  to  see where user accounts start. For example, if UID_MIN is  500, then you would also need to take into account that the unsigned representation of -1 is higher than 500. So you would address this with the following piece of a rule:

       -F auid>=500 -F auid!=4294967295
```

Also, look at some other audit rules examples (e.g. in ```/usr/share/doc/audit/stig.rules```)  suggests that number / "constant" to be used almost in any audit rule.

Comment (jan-cerny) [inline shared/fixes/bash/templates/remediation_functions]: Thanks!

**Comment by isimluk** [comment]
Looks fine to me.

Thanks!

**Pull request merged**